pom.xml

```diff
@@ -176,7 +176,7 @@
         <dep.kafka-clients.version>3.3.2</dep.kafka-clients.version>
         <dep.casandra.version>4.17.0</dep.casandra.version>
         <dep.minio.version>8.5.5</dep.minio.version>
-        <dep.iceberg.version>1.3.1</dep.iceberg.version>
+        <dep.iceberg.version>1.4.0</dep.iceberg.version>
```

Comment:
we also need to bump the matching nessie version while bumping the Iceberg version. 

Please refer the previous Iceberg version bump PR
https://github.com/trinodb/trino/pull/17715/files

Reply:
The CI is failing on this. I think it is best to do the Nessie bump in a separate PR.

Reply:
But they are dependent on each other. Sometimes newer version of Nessie may not work with the Nessie client in older version of the Iceberg. Hence, we used to bump them together.